content/raw/chat/chat_input.md

```diff
@@ -0,0 +1,127 @@
+    1) Results that match the search in the beginning of the username
+    2) Results that closely match the query
+    3) Alphabetical
+5. User sees that the person they wish to @mention is at the top of the suggestions menu (top menu item is always focused by default)
+6. User presses ENTER to select the focused contact
+7. System closes the suggestions menu and completes the @mention string in the text input box
+USE CASE ENDS
+
+*Variation 5a - User moves the focus in the menu*
+1. User sees that the person they wish to @mention in the suggestions menu but that person is not focused 
+2. User moves the focus in the suggestions menu using the up and down cursors to the person they wish to @mention (if needed)
+RETURN TO 7
+
+*Variation 5b - User abandons selecting somebody to @mention*
+1. User decides that they don't want to @mention anybody 
+2. User presses SPACE or ESC
+3. System closes the suggestions menu and returns focus to the text entry box.
+4. System places the text caret at the end of whatever the user has typed, including whatever the user typed while navigating and/or closing the suggestions menu.   
+USE CASE ENDS
+
+### Emojis
+
+TODO
+
+### Code Tags
+
+TODO
+
+### Text Formatter
+
+TODO
+
+## Functional Requirements
+
+**preconditions**
+
+- The user MUST be in a chat
+
+### Mentions
+
+User types a message and when the character `@` is typed it opens a menu with a list of suggestions, the user can then select a suggestion to autocomplete it
+
+- Typing `@` MUST open a suggestions menu
+- Typing after the `@` character will filter the suggestions to match what is being suggested
+- Initially the ordering of the suggestions SHOULD follow an alphabetical order
+- When a user starts the search the ordering of the suggestions menu MUST follow the following criteria in order
+    1. Results that match the search in the beginning of the username
+    2. Results that closely match the query
+    3. Alphabetical
+- The suggestion menu MUST have a selector
+- Pressing the key `Enter` or the key `Tab` MUST autocomplete with the currently selected mention
+
+### Emojis
+
+User types a message and once the character `:` is typed the next character type will either autoreplace the text with an emoji or open a suggestions menu
+
+- Typing `:` followed by a known character that forms an emoji MUST auto replace it with an emoji for example typing `:)` will replace it with 🙂
+- Typing `:` followed by one or more alphabetical characters MUST open a suggestions menu
+- When a user starts the search the ordering of the suggestions menu MUST follow the following criteria in order
+    1. Results that match the search in the beginning of the emoji name
+    2. Results that closely match the query
+    3. Alphabetical
+- The suggestion menu MUST have a selector
+- Pressing the key `Enter` or the key `Tab` MUST autocomplete with the currently selected emoji
+
```

Comment:
- Pressing control + command + space opens native emoji picker (MacOS) I believe there is similar shortcut to this on Windows and Linux.